sos/cleaner/preppers/__init__.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import logging
+
+
+class SoSPrepper():
+    """
+    A prepper is a way to prepare loaded mappings with selected items within
+    an sos report prior to beginning the full obfuscation routine.
+
+    This was previously handled directly within archives, however this is a bit
+    cumbersome and doesn't allow for all the flexibility we could use in this
+    effort.
+
+    Preppers are separated from parsers but will leverage them in order to feed
+    parser-matched strings from files highlighted by a Prepper() to the
+    appropriate mapping for initial obfuscation.
+
+    Preppers may specify their own priority in order to influence the order in
+    which mappings are prepped. Further, Preppers have two ways to prepare
+    the maps - either by generating a list of filenames or via directly pulling
+    content out of select files without the assistance of a parser. A lower
+    priority value means the prepper should run sooner than those with higher
+    values.
```

Comment:
Nitpick comment: swap 2nd and 3rd sentence (as priority levels follows the 1st sentence, while "two ways to prepare the maps" is followe by the next paragraph.